pkg/tc/classes.go

```
@@ -0,0 +1,383 @@
+// Copyright 2012-20124 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package trafficctl
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"strconv"
+
+	"github.com/florianl/go-tc"
+)
+
+const (
+	QFQHelp = `Usage: ... qfq weight NUMBER maxpkt BYTES`
+
+	maxUint32 = 0xFFFF_FFFF
+)
+
+func ParseHFSCClassArgs(out io.Writer, args []string) (*tc.Object, error) {
+	ret := &tc.Object{}
+	var fscOK, rscOK, uscOK bool
+	hfsc := &tc.Hfsc{}
+
+	if len(args) < 2 {
+		return nil, ErrNotEnoughArgs
+	}
+
+	for i := 0; i < len(args); i = i + 2 {
+		switch args[i] {
+		case "rt":
+			sc, err := HFSCGetSC(args[i+1:])
+			if err != nil {
+				return nil, err
+			}
+			hfsc.Rsc = sc
+			rscOK = true
+		case "ls":
+			sc, err := HFSCGetSC(args[i+1:])
+			if err != nil {
+				return nil, err
+			}
+			hfsc.Fsc = sc
+			fscOK = true
+		case "sc":
+			sc, err := HFSCGetSC(args[i+1:])
+			if err != nil {
+				return nil, err
+			}
+			hfsc.Fsc = sc
+			hfsc.Rsc = sc
+			rscOK = true
+			fscOK = true
+		case "ul":
+			sc, err := HFSCGetSC(args[i+1:])
+			if err != nil {
+				return nil, err
+			}
+			hfsc.Usc = sc
+			uscOK = true
+		case "help":
+		default:
+			// WHAT IS THIS??????
```

do we need default this or maybe we need to return an error?